tracking/R3BMDFWrapper.cxx

```diff
@@ -0,0 +1,215 @@
+#include<fstream>
+
+R3BMDFWrapper::R3BMDFWrapper()
+    : mdf_NVariables(0)
+    , pca_NVariables(0)
+    , is_PCA(false)
+{
+}
+
+R3BMDFWrapper::R3BMDFWrapper(const char* mdf_file)
+{
+    InitMDF(mdf_file);
+    is_PCA=false;
+}
+
+R3BMDFWrapper::~R3BMDFWrapper()
+{
+}
+
+//This method will read a special txt file with MDF parameters, don't change
+void R3BMDFWrapper::InitMDF(const char* mdf_file)
+{
+    int i;
+    std::ifstream file(mdf_file);
+    if(file.is_open())
+    {
+        file >> mdf_NVariables >> mdf_NCoefficients >> mdf_NMaxFunctions
+            >> mdf_DMean >> mdf_PolyType;
+
+        
+        mdf_XMean = new double[mdf_NVariables];
+        for(i=0; i<mdf_NVariables; i++)
+            file >> mdf_XMean[i];
+
+        mdf_XMin = new double[mdf_NVariables];
+        for(i=0; i<mdf_NVariables; i++)
+            file >> mdf_XMin[i];
+
+        mdf_XMax = new double[mdf_NVariables];
+        for(i=0; i<mdf_NVariables; i++)
+            file >> mdf_XMax[i];
+
+        mdf_Coefficient = new double[mdf_NCoefficients];
+        for(i=0; i<mdf_NCoefficients; i++)
+            file >> mdf_Coefficient[i];
+
+        mdf_CoefficientRMS = new double[mdf_NCoefficients];
+        for(i=0; i<mdf_NCoefficients; i++)
+            file >> mdf_CoefficientRMS[i];
+
+        mdf_Power = new int[mdf_NCoefficients*mdf_NVariables];
+        for(i=0; i<mdf_NCoefficients*mdf_NVariables; i++)
+            file >> mdf_Power[i];
+        file.close();
+    }
+
+    else 
+        std::cout << "\nUnable to open MDF parameter file!!";
+    return;
+
+}
+
+void R3BMDFWrapper::PrintMDF()
+{
+    std::cout << "\n\n======== Initialized MDF from the R3BMDFWrapper:";
+
+    std::cout << std::setprecision(9) << "\nmdf_NVariables = "  << mdf_NVariables
+        << "\nmdf_NCoefficients = " << mdf_NCoefficients
+        << "\nmdf_NMaxFunctions = " << mdf_NMaxFunctions
+        << "\nmdf_DMean = " << mdf_DMean 
+        << "\nmdf_PolyType = " << mdf_PolyType;
+
+    std::cout << "\nmdf_Xmean = ";
+    for(int i=0; i<mdf_NVariables; i++)
+        std::cout << mdf_XMean[i] << ", ";
+
+    std::cout << "\nmdf_Xmin = ";
+    for(int i=0; i<mdf_NVariables; i++)
+        std::cout << mdf_XMin[i] << ", ";
+
+    std::cout << "\nmdf_Xmax = ";
+    for(int i=0; i<mdf_NVariables; i++)
+        std::cout << mdf_XMax[i] << ", ";
+
+    std::cout << "\nmdf_Coefficient = \n";
+    for(int i=0; i<mdf_NCoefficients; i++)
+        std::cout << mdf_Coefficient[i] << std::endl;
+
+    std::cout << "\nmdf_CoefficientRMS = \n";
+    for(int i=0; i<mdf_NCoefficients; i++)
+        std::cout << mdf_CoefficientRMS[i] << std::endl;
+
+    std::cout << "\nmdf_Power = \n";
+    for(int i=0; i<mdf_NCoefficients*mdf_NVariables; i++)
+        std::cout << mdf_Power[i] << ", ";
+
+    std::cout << "\n\n";
+}
+
+//This method will read a special txt file with PCA parameter, don't change
+void R3BMDFWrapper::InitPCA(const char* pca_file)
+{
+    int i;
+    std::ifstream file(pca_file);
```

Comment:
Again, it would be preferible to have all in parameter containers

Reply:
It's ok for me... do you want to change first the LOG vs cout issue and correct the typo? Then we will accept the PR...

Reply:
Yes, working on it right now :) I will make soon a new pull request where the typo and LOG issues are fixed